src/compiler/scala/tools/nsc/typechecker/RefChecks.scala

```diff
@@ -1451,9 +1446,9 @@ abstract class RefChecks extends Transform {
             )
       }
 
-      sym.isSourceMethod &&
+      sym.name == nme.apply &&
+        !(sym hasFlag STABLE) && // ???
```

Comment:
To clarify, I just inlined `isSourceMethod` here, acting on the TODO in its doc comment.